src/meta/processors/parts/CreateSpaceAsProcessor.cpp

```diff
@@ -0,0 +1,230 @@
+/* Copyright (c) 2018 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/processors/parts/CreateSpaceAsProcessor.h"
+
+#include "meta/ActiveHostsMan.h"
+
+namespace nebula {
+namespace meta {
+
+void CreateSpaceAsProcessor::process(const cpp2::CreateSpaceAsReq &req) {
+  SCOPE_EXIT {
+    if (rc_ != nebula::cpp2::ErrorCode::SUCCEEDED) {
+      handleErrorCode(rc_);
+    }
+    onFinished();
+  };
+
+  folly::SharedMutex::WriteHolder wHolder(LockUtils::spaceLock());
+  auto oldSpaceName = req.get_old_space_name();
+  auto newSpaceName = req.get_new_space_name();
+  auto oldSpaceId = getSpaceId(oldSpaceName);
+  auto newSpaceId = getSpaceId(newSpaceName);
+
+  if (!nebula::ok(oldSpaceId)) {
+    rc_ = nebula::error(oldSpaceId);
+    LOG(ERROR) << "Create Space [" << newSpaceName << "] as [" << oldSpaceName
+               << "] failed. rc = " << apache::thrift::util::enumNameSafe(rc_);
+    return;
+  }
+
+  if (nebula::ok(newSpaceId)) {
+    if (!req.get_if_not_exists()) {
+      LOG(ERROR) << "Create Space " << newSpaceName << " as " << oldSpaceName << "failed. "
+                 << newSpaceName << " already exist";
+      rc_ = nebula::cpp2::ErrorCode::E_EXISTED;
+      return;
+    }
```

Comment:
When` req.get_if_not_exists()` is true, return success directly？
But this new space is not the desired space anymore.
Maybe `if_not_exists `is not needed?

Reply:
good catch, removed.